config/main.py

```diff
@@ -2490,11 +2492,11 @@ def add_snmp_agent_address(ctx, agentip, port, vrf):
             for ipaddr in ipaddresses[ip_family[ip.version]]:
                 if agentip == ipaddr['addr']:
```

Comment:
>ipaddr['addr'] [](http://example.com/codeflow?start=30&length=14)

Currently the ConfigDB has no way to enforce the lowercase for SNMP_AGENT_ADDRESS_CONFIG lowercase key. So there may be already lowercase entries there.

Suggest you compare ipv6 address case insensitive. This library may help [https://docs.python.org/3/library/ipaddress.html](https://docs.python.org/3/library/ipaddress.html)

Reply:
I see, it's better to compare the lowercase of IPv6 addresses, and I have modified my PR, thanks.